README.md

```diff
@@ -5,3 +5,39 @@ A file can fall into one of three categories.
 1. Files that are core to Salt and will remain in the Salt codebase
 2. Files that will be moved out of the Salt codebase but continue to be maintained by the Salt Core Team
 3. Files that will be moved out into a dump repository and will no longer be maintained by the Salt Core Team
+
+
+We'll use `git filter-repo` to migrate modules to a separate repositoy while
+retaining their git history. Once the migration has taken place, we'll do a
+`git rm` against the main salt repository.
+
+
+Install filter-repo (Debian)
+```
+sudo apt-get install -y git-filter-repo
+```
+
+Now make a fresh clone and run filter repo. Starting in the root of this repository.
+
+```
+FILESDIR=$(pwd)
+cd /tmp
+git clone git@github.com:saltstack/salt.git community-extensions
+cd community-extensions
+git filter-repo --dry --paths-from-file=$FILESDIR/community-ext-modules.txt
+```
+
+If eveything looks good... Send it!
+
+```
+git filter-repo --paths-from-file=$FILESDIR/community-ext-modules.txt
+```
+
+Now we can remove the migrated files from the salt repo.
+
+```
+cd ~/src/salt
```

Comment:
Not everyone keeps their Salt in ~/src/salt, how about 

Now we can remove the migrated files from your copy of the Salt repo, for example: if it is located in ~/src/salt

Reply:
This is meant to be an example with enough information that someone could figure it out. Not a step by step guide.

Reply:
I think you need to treat as a step by step guide. It will cut down on number of issues, if we do.

Reply:
I think it's fine. People will understand that's where the salt repo is. Even I understood it...